File: CS/E1231/DXExample.Module/UserResource.cs

using DevExpress.Xpo;

using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;

namespace DXExample.Module {
    [DefaultClassOptions]
    public class UserResource : BaseObject {
        public UserResource(Session session) : base(session) { }
        private string _ResourceName;
        public string ResourceName {
            get { return _ResourceName; }
            set { SetPropertyValue("ResourceName", ref _ResourceName, value); }
        }
        [Association("Users-Resources")]
        public XPCollection<CustomUser> Owners {
            get { return GetCollection<CustomUser>("Owners"); }
        }
    }

}

File: VB/VB/DXExample.Module/UserResource.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl

Namespace DXExample.Module
	<DefaultClassOptions> _
	Public Class UserResource
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _ResourceName As String
		Public Property ResourceName() As String
			Get
				Return _ResourceName
			End Get
			Set(ByVal value As String)
				SetPropertyValue("ResourceName", _ResourceName, value)
			End Set
		End Property
		<Association("Users-Resources")> _
		Public ReadOnly Property Owners() As XPCollection(Of CustomUser)
			Get
				Return GetCollection(Of CustomUser)("Owners")
			End Get
		End Property
	End Class

End Namespace
